modules/CircularLayout/src/main/java/org/gephi/layout/plugin/circularlayout/circlelayout/CircleLayout.java

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright (c) 2010, Matt Groeninger
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification, are
+ * permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this list of
+ * conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list
+ * of conditions and the following disclaimer in the documentation and/or other materials
+ * provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.gephi.layout.plugin.circularlayout.circlelayout;
+
+import java.util.*;
+import org.gephi.graph.api.*;
+import org.gephi.graph.spi.LayoutData;
+import org.gephi.layout.plugin.circularlayout.layouthelper.*;
+import org.gephi.layout.spi.Layout;
+import org.gephi.layout.spi.LayoutBuilder;
+import org.gephi.layout.spi.LayoutProperty;
+import org.openide.util.Lookup;
+import org.openide.util.NbBundle;
+
+/**
+ *
+ * @author Matt
+ */
+public class CircleLayout extends LayoutHelper implements Layout
+{
+   private Graph       graph;
+   private double      diameter;
+   private boolean     boolfixeddiameter;
+   private String      strNodeplacement;
+   static final double TWO_PI = (2 * Math.PI);
+
+   public CircleLayout(LayoutBuilder layoutBuilder, double diameter, boolean boolfixeddiameter)
+   {
+      super(layoutBuilder);
+      this.diameter          = diameter;
+      this.boolfixeddiameter = boolfixeddiameter;
+   }
+
+   @Override
+   public void initAlgo()
+   {
+      this.setConverged(false);
+      this.graph = graphModel.getGraphVisible();
+      float[]        nodeCoords  = new float[2];
+      double         tmpcirc     = 0;
+      double         tmpdiameter = 0;
+      int            index       = 0;
+      int            nodecount   = this.graph.getNodeCount();
+      double         noderadius  = 0;
+      double         theta       = TWO_PI / nodecount;
+      double         lasttheta   = 0;
+      TempLayoutData posData;
+
+      Node[] nodes = this.graph.getNodes().toArray();
```

Comment:
You can safely remove the toArray() here if you use the readLock/Unlock I've mentioned earlier. This would otherwise an unnecessary copy of the nodes. Accessing nodes through `graph.getNodes()` is very fast in Gephi 0.9.0 and there's no perf. benefits of copying to an array, even if you read nodes multiple times.